daemon/errors.go

```diff
@@ -14,16 +14,12 @@ func (d *Daemon) imageNotExistToErrcode(err error) error {
 			e := fmt.Errorf("No such image: %s", dne.RefOrID)
 			return errors.NewRequestNotFoundError(e)
 		}
-		tag := reference.DefaultTag
-		ref, err := reference.ParseNamed(dne.RefOrID)
+		ref, err := reference.ParseNormalizedNamed(dne.RefOrID)
```

Comment:
I think we should change the `RefOrID` to `reference.Reference` instead of reparsing here.